### PR TITLE
convert lifecycle action object to map

### LINKED ISSUE #51: lifecycle_rules not working for action.type=delete
_There are two issues but they are both closely related_

**Steps to repeat issue**
Add the following block for the module:
``` 
lifecycle_rules = [{
    action = {
      type = "Delete"
    }
    condition = {
      age = 1
    }
  }]
```

Error occurs when planning or applying:
```  Error: Invalid value for module argument
  on .terraform/somepathhere/main.tf line 42, in module "my-bucket":
  42:   lifecycle_rules = [{
  43:     action = {
  44:       type = "Delete"
  46:     }
  47:     condition = {
  48:       age = 1
  49:     }
  50:   }]
The given value is not suitable for child module variable "lifecycle_rules"
defined at
.terraform/modules/somepathhere/terraform-google-cloud-storage-1.5.0/variables.tf:130,1-27:
element 0: attribute "action": attribute "storage_class" is required.
```

When I apply the required storage_class to the module:
```
lifecycle_rules = [{
    action = {
      type = "Delete"
      storage_class = "STANDARD"
    }
    condition = {
      age = 1
    }
  }]
```

`terraform plan` will work,  but on a `terraform apply` we get another error:
```
Error: googleapi: Error 400: Invalid argument, invalid
  on .terraform/modules/somepathhere/main.tf line 21, in resource "google_storage_bucket" "buckets":
  21: resource "google_storage_bucket" "buckets" {
```

**Workaround**
I modified the local copy of the module to ignore the storage_class and I was able to run both a `plan` and `apply` with no issues.

### PR DESCRIPTION
**Pull request by umairidris**
Fix #51 

Also clean up simple bucket example to be clearer.

### CODE REVIEW

File: examples/simple_bucket/variables.tf [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "project_id" {
+  description = "The ID of the project in which to provision resources."
+  type        = string
+}
+
+variable "name" {
```
Comment (morgante) [inline examples/simple_bucket/variables.tf:22]: We should hardcode this or provide a default.

Comment (umairidris) [reply examples/simple_bucket/variables.tf:22]: If we hard code it won't it have a conflict? Since buckets are globally unique

Comment (morgante) [reply examples/simple_bucket/variables.tf:22]: Hmm, fair point.

**Comment by morgante** [comment]
@umairidris Could you push again? It looks like CI is hanging and needs to be triggered.

**Comment by umairidris** [comment]
> @umairidris Could you push again? It looks like CI is hanging and needs to be triggered.

Done.

**Pull request merged**